awx/api/generics.py

```diff
@@ -208,19 +209,37 @@ def finalize_response(self, request, response, *args, **kwargs):
             return response
 
         if response.status_code >= 400:
-            status_msg = "status %s received by user %s attempting to access %s from %s" % (
-                response.status_code,
-                request.user,
-                request.path,
-                request.META.get('REMOTE_ADDR', None),
-            )
+            try:
+                status_msg = getattr(settings, 'API_400_ERROR_LOG_FORMAT').format(
+                    status_code=response.status_code,
+                    user_name=request.user,
+                    url_path=request.path,
+                    remote_addr=request.META.get('REMOTE_ADDR', None),
+                    error=response.data.get('error', response.status_text),
+                )
+            except Exception as e:
+                if getattr(settings, 'API_400_ERROR_LOG_FORMAT', None):
+                    logger.error("Unable to format API_400_ERROR_LOG_FORMAT setting, defaulting log message")
+                status_msg = (
+                    settings_registry.get_setting_field('API_400_ERROR_LOG_FORMAT')
+                    .get_default()
+                    .format(
+                        status_code=response.status_code,
+                        user_name=request.user,
+                        url_path=request.path,
+                        remote_addr=request.META.get('REMOTE_ADDR', None),
+                        error=response.data.get('error', None),
```

Comment:
So you don't repeat yourself, I'd prefer to dump these parameters into a dict before-hand. The `error` value is different in the 2 cases.